src/scripts/utils.py

```diff
@@ -0,0 +1,399 @@
+"""Slurp migration pipeline: utils"""
+import os
+import pickle
+import subprocess
+from datetime import datetime
+from glob import glob
+from typing import Dict, List, Set, Union
+
+import curies
+import pandas as pd
+from jinja2 import Template
+from oaklib import OntologyResource
+from oaklib.implementations import ProntoImplementation, SqlImplementation
+from oaklib.interfaces.basic_ontology_interface import RELATIONSHIP
+from oaklib.types import CURIE, URI
+from pandas.errors import EmptyDataError
+
+PREFIX = str
+TRIPLE = RELATIONSHIP
+SCRIPTS_DIR = os.path.dirname(os.path.realpath(__file__))
+PROJECT_DIR = os.path.join(SCRIPTS_DIR, '..', '..')
+ONTOLOGY_DIR = os.path.join(PROJECT_DIR, 'src', 'ontology')
+TEMP_DIR = os.path.join(ONTOLOGY_DIR, 'tmp')
+CACHE_DIR = TEMP_DIR
+
+
+# todo: there are remaining todo's in this class
+class Term:
+    """A lightweight class representing an ontological (e.g. RDF/OWL) term class."""
+
+    # todo: I have heard you're not supposed to default to empty list, but doing this is better UX
+    #  ...because don't need to handle possible "TypeError: 'NoneType' object is not iterable". However, I might
+    #  ...want to re-look into why this is advised not to do.
+    def __init__(
+        self, uri: URI = None, curie: CURIE = None, label: str = None, definition: str = None,
+        direct_owned_parent_uris: List[Union[URI, CURIE]] = [],
+        direct_owned_parent_curies: List[Union[URI, CURIE]] = [],
+        ontology: ProntoImplementation = None, owned_prefix_map: Dict[PREFIX, URI] = None,
+        general_prefix_map: Dict[PREFIX, URI] = None
+    ):
+        """Initialize a term.
+
+        ontology: Requires use of `oaklib`, particularly the `ProntoImplementation`.
+        owned_prefix_map: A prefix_map that includes *only* of the prefixes 'owned' by the ontology. Used to
+        general_prefix_map: A general prefix_map. For conversion of URIs and CURIEs, even if not owned by ontology.
+        direct_owned_parent_uris: Some items will be CURIE if URI not available and conversion fails
+        direct_owned_parent_curies: Some items will be URI if CURIE not available and conversion fails"""
+        if not (uri or curie):
+            raise ValueError('Term: Must supply either `uri` or `curie`.')
+        self.label = label
+        self.definition = definition
+        self.ontology = ontology
+        self.owned_prefix_map = owned_prefix_map
+        self.general_prefix_map = general_prefix_map
+        if owned_prefix_map or general_prefix_map:
+            self._converter = curies.Converter.from_prefix_map(
+                general_prefix_map if general_prefix_map else owned_prefix_map)
+        self.uri = uri
+        self.curie = curie
+        self.direct_owned_parent_uris = direct_owned_parent_uris
```

Comment:
@matentzn **Why OMIM parents are (correctly) not showing up**
I took a look, and it looks like the way we have `omim.owl` does not have any parents that are 'owned' by OMIM. I'm assuming this is not an issue with OMIM itself or our OMIM issue, and thus not an issue for us. But I want you to review and check to be sure.

Example OMIM class:
```owl
    <owl:Class rdf:about="https://omim.org/entry/100650">
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/SO_0000704"/>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/RO_0002525"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/CHR_9606chr12q24.12"/>
            </owl:Restriction>
        </rdfs:subClassOf>
```

As you can see, it has parents, but they're from `SO`, `RO`, and `CHR`. As they are not 'owned' by the ontology, they will not appear int he slurp output.

Reply:
We looked into this and decided that we will [add genes to the exclusion table](https://github.com/monarch-initiative/mondo-ingest/issues/82).